lib/utils/aws/region.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"regexp"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws/endpoints"
```

Comment:
I was about to recommend the usage of sdk v2, but I'm not sure they provide the same capabilities for this use case.
There's a https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2#Client.DescribeRegions
But you need to specify a region beforehand 🤷 

However, there's also this file that should give us the latest known list of regions:
https://github.com/aws/aws-sdk-go-v2/blob/main/codegen/smithy-aws-go-codegen/src/main/resources/software/amazon/smithy/aws/go/codegen/endpoints.json
https://raw.githubusercontent.com/aws/aws-sdk-go-v2/master/codegen/smithy-aws-go-codegen/src/main/resources/software/amazon/smithy/aws/go/codegen/endpoints.json
We could use that to re-generate the list (eg using a make target, as a dependency for the release target), instead of relying on the current sdk version.

To detect unknown regions, we can/should use the multiple regex (converged into one, if possible) that you can also find in the file above.

Reply:
> There's a https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2#Client.DescribeRegions

If my understanding is correct,`ec2:DescribeRegions` is an API call which requires extra IAM permission, and it only returns the regions that's enabled for the caller.

> We could use that to re-generate the list (eg using a make target, as a dependency for the release target), instead of relying on the current sdk version.

ya, I think there are many ways we can generate the list is difficult with a make target. I kinda want to avoid this tho since a customer has to upgrade their binaries to get the list. What do you think?

> To detect unknown regions, we can/should use the multiple regex (converged into one, if possible) that you can also find in the file above.

Are you referring to the regex like `^(us|eu|ap|sa|ca|me|af|il)\\-\\w+\\-\\d+$`? `il`, for example, is the new one getting added recently so i don't think these regexes are future proof when new regions are opened in a new country.

Reply:
> ya, I think there are many ways we can generate the list with a make target. I kinda want to avoid this tho since a customer has to upgrade their binaries to get the list. What do you think?

Sure :+1: 

As for the regex, let's at least be less strict than their current regex.
For instance, I don't think we should use the geo direction strings